Modules/_ctypes/_ctypes.c

```diff
@@ -1405,13 +1405,30 @@ PyCArrayType_new(PyTypeObject *type, PyObject *args, PyObject *kwds)
     type_attr = NULL;
 
     length_attr = PyObject_GetAttrString((PyObject *)result, "_length_");
-    if (!length_attr || !PyLong_Check(length_attr)) {
-        PyErr_SetString(PyExc_AttributeError,
-                        "class must define a '_length_' attribute, "
-                        "which must be a positive integer");
-        Py_XDECREF(length_attr);
+    if (!length_attr) {
+        if (!PyErr_Occurred() ||
+            PyErr_ExceptionMatches(PyExc_AttributeError))
```

Comment:
It is worth to add similar check for `_type_`.

Reply:
I'll make a new PR for that. Should I also make a new issue?